pilot/pkg/config/kube/crd/controller/client.go

```diff
@@ -409,6 +417,22 @@ func (cl *Client) objectInEnvironment(o *model.Config) bool {
 	return configEnv == cl.revision
 }
 
+// CRDExists checks if a CRD exists in the cluster or not.
+func (cl *Client) CRDExists(schema collection.Schema) (bool, error) {
```

Comment:
I remember once upon a time, someone removed the crd register from pilot.

You add back again?

Reply:
This is not registering the crd, it's instead making a missing crd not block startup. If the crd is later added, then as watch for that crd

We could remove the code to watch new CRDs if its too complicated and require a restart I guess